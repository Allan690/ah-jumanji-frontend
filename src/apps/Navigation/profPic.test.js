import React from 'react';

import Enzyme, { shallow, mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

import { MemoryRouter } from 'react-router-dom';
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import ProfPic from './profPic';

Enzyme.configure({ adapter: new Adapter() });
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('<ProfPic />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<ProfPic debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('ProfPic', () => {
  it('should render self and subcomponents', () => {
    const store = mockStore({});
    const enzymeWrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ProfPic />
        </MemoryRouter>
      </Provider>,);
    expect(enzymeWrapper.find('ProfPic')).toBeDefined();
    expect(enzymeWrapper.find('button')).toHaveLength(1);
  });
});
