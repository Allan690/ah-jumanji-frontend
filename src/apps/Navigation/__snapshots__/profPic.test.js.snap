// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfPic /> should match the snapshot 1`] = `
<button
  aria-expanded="false"
  aria-haspopup="true"
  className="nav-link dropdown-toggle btn-link"
  data-toggle="dropdown"
  id="navbarDropdown"
  type="button"
>
  <img
    alt="avatar"
    className="followers-avatar navbar-avatar rounded-circle"
    src="https://cdn-images-1.medium.com/max/1200/1*MccriYX-ciBniUzRKAUsAw.png"
  />
</button>
`;
